synapse/logging/scopecontextmanager.py

```diff
@@ -1,171 +0,0 @@
-# Copyright 2019 The Matrix.org Foundation C.I.C.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.import logging
-
-import logging
-from types import TracebackType
-from typing import Optional, Type
-
-from opentracing import Scope, ScopeManager, Span
-
-import twisted
-
-from synapse.logging.context import (
-    LoggingContext,
-    current_context,
-    nested_logging_context,
-)
-
-logger = logging.getLogger(__name__)
-
-
-class LogContextScopeManager(ScopeManager):
-    """
-    The LogContextScopeManager tracks the active scope in opentracing
-    by using the log contexts which are native to synapse. This is so
-    that the basic opentracing api can be used across twisted defereds.
-
-    It would be nice just to use opentracing's ContextVarsScopeManager,
-    but currently that doesn't work due to https://twistedmatrix.com/trac/ticket/10301.
```

Comment:
https://github.com/twisted/twisted/issues/10301 seems to be fixed now (via https://github.com/twisted/twisted/pull/1690). How do we use this? And can we use this for OTEL stuff?

Reply:
We still can't use it -- we support versions of Twisted which don't include the fix there.

Reply:
including, sadly, the most recent twisted release.

contextvars have enough potential to improve our lives that I might be prepared to risk the wrath of packagers by requiring the latest twisted release, but right now even that's an impossible argument.

Reply:
We need that fix for this OpenTelemetry stuff, https://github.com/matrix-org/synapse/pull/13400#discussion_r936195492 (details). There is no concept of scope managers in OTEL, it just uses `contextvars and the latest Twisted fixes it.

Can we require the latest version of Twisted for the tracing stuff to work at least?

Reply:
> Can we require the latest version of Twisted for the tracing stuff to work at least?

From the [backend chapter sync last month](https://docs.google.com/document/d/1kmGRzPFfg_gRY6l0sxjYkSLW6UpMFn9ELQX5CtTLWlA/edit#bookmark=id.5lznznqy01zg), it was discussed that it would probably be fine to require the latest Twisted for tracing as it's mostly a tool for us.